services/supervisor/tests/simple.rs

```diff
@@ -587,3 +660,88 @@ fn several_service_config_changes() {
 
     assert_eq!(actual_params, "Change 4");
 }
+
+#[test]
+fn discard_config_propose_from_auditor() {
+    let mut testkit = TestKitBuilder::auditor()
+        .with_validators(2)
+        .with_rust_service(SimpleSupervisor)
+        .create();
+
+    let cfg_change_height = Height(5);
+    let old_consensus_config = testkit.consensus_config();
+    // Attempt to add ourselves into validators list.
+    let new_consensus_config = {
+        let mut cfg = testkit.consensus_config();
+        cfg.validator_keys.push(testkit.us().public_keys());
+        cfg
+    };
+
+    let old_validators = testkit.network().validators();
+
+    // Sign request by an auditor.
+    let keys = &testkit.network().us().service_keypair();
+    let config_propose = ConfigPropose::actual_from(cfg_change_height)
+        .consensus_config(new_consensus_config.clone())
+        .sign_for_simple_supervisor(keys.0, &keys.1);
+
+    testkit.create_block_with_transaction(config_propose);
```

Comment:
Nit: It could make sense to check that the transaction has errored here, and to verify the error code.